_posts/2023-07-10-custom-http-signaling-notification.adoc

```diff
@@ -0,0 +1,252 @@
++++<!-- more -->+++
+
+== Custom Signal & Notification Channel
+
+In Debezium, it is possible to customize the signal and notification channels to fit specific requirements. For example, we can achieve a customization by creating an `HTTP` channel for both signals and notifications. This `HTTP` channel receives signals from an http endpoint, and notifications can be sent back to the endpoint upon the delivery of the signal.
+
+Let's explore an example that demonstrates how to create and utilize the `HTTP` signal and notification channels using the Debezium Postgres connector, a https://www.mock-server.com/[Mock Server] to send signals, and https://www.toptal.com/developers/postbin/[Postbin] to receive notifications via http endpoint.
+
+=== Setting up the `HTTP` Signal Channel:
+
+- Configure the Debezium Postgres connector to receive signals whenever a relevant database change occurs.
+- Set up a service to send signals to Debezium using the `HTTP` channel. The service can be a database, a third-party application, or any other system that can send http requests. In this example, we will use the Mock Server to send signals to Debezium. The Mock Server is a service that can be used to mock http requests and responses.
+- Configure the Mock Server to send the signals via a http endpoint using the appropriate HTTP method (e.g., POST).
+- Customize the `HTTP` channel settings to define the http endpoint URL, authentication, headers, and any additional parameters as needed.
+
+=== Setting up the `HTTP` Notification Channel:
+
+- Once the signal is received and processed by Debezium, it can trigger the posting of a notification to a http endpoint. In this example, we will send the notification to the Postbin bin using the `HTTP` channel. Postbin is a service that can be used to receive http requests and view the request details.
+- Customize the `HTTP` channel settings for notifications, create a bin in Postbin, and define the http endpoint URL, authentication, headers, and any additional parameters as needed.
+- Forward the notification event to the http endpoint i.e Postbin bin using the appropriate HTTP method (e.g., POST). The notification payload can be customized as needed.
+
+The complete source code for this example in the blog post is provided in the Debezium examples repository under https://github.com/debezium/debezium-examples/tree/main/http-signaling-notification[http-signal-notification] directory.
+
+Create a java project to build the `HTTP` signal and notification channels. Run the following command to create a new java project using Maven:
+
+[source,shell]
+----
+mvn archetype:generate
+    -DgroupId=io.debezium.examples
+    -DartifactId=http-signaling-notification
+----
+
+Add the following dependency to the `pom.xml` file with the Debezium version (2.3 and later versions):
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.debezium</groupId>
+    <artifactId>debezium-core</artifactId>
+    <version>${debezium.version}</version>
```

Comment:
```suggestion
    <version>2.3.0.Final</version>
```
It's generally better to provide bits that can be copied directly without variables for clarity.